ldms/src/core/ldms.h

```diff
@@ -1336,6 +1337,36 @@ uint64_t ldms_set_meta_gn_get(ldms_set_t s);
  * \returns	The 64bit data generation number.
  */
 uint64_t ldms_set_data_gn_get(ldms_set_t s);
+
+/**
+ * \brief Get the heap generation number.
+ */
+uint64_t ldms_set_heap_gn_get(ldms_set_t s);
+
+/**
+ * \brief Tell LDMS to copy previous data in the set array on transaction begin.
+ *
+ * When \c ldms_transaction_begin() is called, the set data points to the next
+ * data slot in the set array and the metric modification by the application
+ * will be applied to the new data slow. By default, the new data slot is left
+ * as-is is.
```

Comment:
May add "This presumes that the sample() function will update all list entry data at each invocation."